### PR TITLE
Add basic cli

### LINKED ISSUE #2: Add cli
we should add a CLI using the library and use Quarkus to built it so we should have in java folder api for the library and cli for the CLI

Reference https://github.com/lstocchi/test-recognizer/issues/3

### PR DESCRIPTION
**Pull request by lstocchi**
it resolves #2

### CODE REVIEW

File: cli/pom.xml [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <parent>
+    <artifactId>alizer</artifactId>
+    <groupId>com.redhat.devtools.alizer</groupId>
+    <version>0.1.0-SNAPSHOT</version>
+  </parent>
+  <modelVersion>4.0.0</modelVersion>
+  <artifactId>cli</artifactId>
+  <properties>
+    <compiler-plugin.version>3.8.1</compiler-plugin.version>
```
Comment (jeffmaury) [inline cli/pom.xml:12]: Not sure the non quarkus properties are required as they are probably herited from jboss-parent, to be checked

File: cli/pom.xml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    </dependency>
+    <dependency>
+      <groupId>io.quarkus</groupId>
+      <artifactId>quarkus-junit5</artifactId>
+      <scope>test</scope>
+    </dependency>
+      <dependency>
+          <groupId>com.redhat.devtools.alizer</groupId>
+          <artifactId>api</artifactId>
+          <version>0.1.0-SNAPSHOT</version>
```
Comment (jeffmaury) [inline cli/pom.xml:52]: Use ${project.version}

File: cli/src/main/java/com/redhat/devtools/recognizer/cli/AlizerCli.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    @Override
+    public void run() {
+        if (name != null) {
+            try {
+                new LanguageRecognizerBuilder().build().analyze(name).forEach(
+                        lang -> {
+                            System.out.printf("%-10s %-10s %-10s %-10s\n",
+                                    lang.getName(),
+                                    (!lang.getFrameworks().isEmpty() ? lang.getFrameworks().get(0) : ""),
```
Comment (jeffmaury) [inline cli/src/main/java/com/redhat/devtools/recognizer/cli/AlizerCli.java:36]: Better display all framework with String.join

File: cli/src/main/java/com/redhat/devtools/recognizer/cli/AlizerCli.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+                                    (!lang.getTools().isEmpty() ? lang.getTools().get(0) : ""),
```
Comment (jeffmaury) [inline cli/src/main/java/com/redhat/devtools/recognizer/cli/AlizerCli.java:37]: Idem

**Review by jeffmaury — CHANGES_REQUESTED** [review]
Also I tried to run the native binary:

./cli-0.1.0-SNAPSHOT-runner .
java.lang.NullPointerException
	at java.io.Reader.<init>(Reader.java:167)
	at java.io.InputStreamReader.<init>(InputStreamReader.java:113)
	at org.apache.commons.io.IOUtils.copy(IOUtils.java:921)
	at org.apache.commons.io.IOUtils.toString(IOUtils.java:2681)
	at org.apache.commons.io.IOUtils.toString(IOUtils.java:2661)
	at com.redhat.devtools.recognizer.api.LanguageFileHandler.initLanguages(LanguageFileHandler.java:46)
	at com.redhat.devtools.recognizer.api.LanguageFileHandler.<init>(LanguageFileHandler.java:34)
	at com.redhat.devtools.recognizer.api.LanguageFileHandler.get(LanguageFileHandler.java:39)
	at com.redhat.devtools.recognizer.api.LanguageRecognizerImpl.analyze(LanguageRecognizerImpl.java:38)
	at com.redhat.devtools.recognizer.cli.AlizerCli.run(AlizerCli.java:32)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1939)
	at picocli.CommandLine.access$1300(CommandLine.java:145)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2352)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2346)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2311)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
	at io.quarkus.picocli.runtime.PicocliRunner$EventExecutionStrategy.execute(PicocliRunner.java:26)
	at picocli.CommandLine.execute(CommandLine.java:2078)
	at io.quarkus.picocli.runtime.PicocliRunner.run(PicocliRunner.java:39)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:119)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:66)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:42)
	at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:30)

**Review by jeffmaury — CHANGES_REQUESTED** [review]
Tested it but native mode is working in a degraded mode. If you go to cli/target and run cli-0.1.0-SNAPSHOT-runner then you will get Java 100%. If you run java -jar cli-0.1.0-SNAPSHOT-runner.jar you will get Java Quarkus Maven 100%

See https://quarkus.io/guides/all-config#quarkus-core_quarkus.native.auto-service-loader-registration

**Comment by lstocchi** [comment]
> Tested it but native mode is working in a degraded mode. If you go to cli/target and run cli-0.1.0-SNAPSHOT-runner then you will get Java 100%. If you run java -jar cli-0.1.0-SNAPSHOT-runner.jar you will get Java Quarkus Maven 100%
> 
> See https://quarkus.io/guides/all-config#quarkus-core_quarkus.native.auto-service-loader-registration

I did the tests before to push and i was convinced it was working fine lol ... better to stop for today :laughing: BTW with that property works fine, thanks!!

**Review by jeffmaury — APPROVED** [review]
LGTM

**Pull request merged**